CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -1316,15 +1316,22 @@ _CFThreadRef _CFThreadCreate(const _CFThreadAttributes attrs, void *_Nullable (*
     return thread;
 }
 
-CF_SWIFT_EXPORT void _CFThreadSetName(const char *_Nullable name) {
+CF_SWIFT_EXPORT int _CFThreadSetName(pthread_t thread, const char *_Nonnull name) {
 #if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_EMBEDDED_MINI
-    pthread_setname_np(name);
+    if (pthread_equal(pthread_self(), thread)) {
+        return pthread_setname_np(name);
+    }
+    return EINVAL;
 #elif DEPLOYMENT_TARGET_LINUX
-    pthread_setname_np(pthread_self(), name);
+    // pthread_setname_np will fail if name >= 16 characters
+    char short_name[16];
```

Comment:
This is technically taking a string which we assume to be constant, and then passing it to the call pointing to a locally allocated buffer. Do we know that the `pthread_setname_np` call is going to take a copy of that string, as opposed to just referring to it by pointer? The call appears to be going to a `PRCTL(PR_SETNAME)` in the Linux implementation; do we know that does the right thing?

I think it would be better to have an assert and return the error message if the thread name is too long, rather than trying to do the defensive copy here.

Reply:
Does `const` say anything about the lifetime of a pointer? What happens if the `String` held in `Thread` type is deallocated before the thread terminates?

I was being defensive because setting thread names seems to be mostly for debugging. An `assert` might be a bit excessive, however the copy could just be removed completely and the call allowed to fail with the `ERANGE` that `pthread_setname_np` would normally return.

Reply:
If we're changing this to return an `Exxx` code, I'd say let the caller deal with it and have the value return an `ERANGE` (or whatever the normal point is). This also permits future library versions to extend the range change, say, up to 255 bytes (whereas you'd be trimming to 16 all the time).